agent/client/connection_set.go

```diff
@@ -0,0 +1,65 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package client contains business logic of working with pmm-managed.
+package client
+
+import "time"
+
+type connectionSet struct {
+	events       []connectionEvent
+	windowPeriod time.Duration
+}
+
+type connectionEvent struct {
+	t         time.Time
+	connected bool
+}
+
+func NewConnectionSet(windowPeriod time.Duration) *connectionSet {
+	return &connectionSet{
+		windowPeriod: windowPeriod,
+	}
+}
+
+func (c *connectionSet) Set(timestamp time.Time, connnected bool) {
+	c.deleteOldEvents()
```

Comment:
I think calling `deleteOldEvents` on each `Set` call will decrease the performance of ping pong messages in case of multiple reconnects

Reply:
Moved calling of it to goroutine